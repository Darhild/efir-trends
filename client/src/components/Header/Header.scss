@import './../../variables';

.Header {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    width: 100%;
    height: 48px;
    padding: 0 16px;

    @media(min-width: $screen-s) {
        justify-content: flex-start;
        height: 66px;
        padding: 14px 16px;
    }

    @media(min-width: $screen-m) {
        padding: 14px 38px;
    }

    &-Logo {
        font-weight: 300;        
        font-size: 28px;
        line-height: 16px;

        @media(min-width: $screen-s) {
            margin-right: 16px;
        }
    }

    .Ya-Logo {
        font-weight: 500;
    }

    &-Search {
        position: absolute;
        right: 0;
        display: flex;
        align-items: center;
        width: auto;
        height: 100%;
        margin-right: 16px;

        @media(min-width: $screen-s) {
            position: relative;
            right: initial;
        }
    }

    &-Authorization {
        margin-left: auto;
    }
}
