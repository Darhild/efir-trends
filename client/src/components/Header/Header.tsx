import React, { Component } from 'react';
import './Header.scss';
import Icon from '../Icon/Icon';
import  { ReactComponent as Search }  from '../../images/svg/search.svg';

class Header extends Component {
    render() {
        return (
            <div className="Header">
                <div className="Header-Logo">
                    <span className='Ya-Logo'>Яндекс</span> Эфир
                </div>
                <Icon><Search /></Icon>
            </div>
        );
    }
}

export default Header;