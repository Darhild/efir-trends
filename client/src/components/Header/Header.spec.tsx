import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';

test('subscribe Header default state', () => {
    const component = renderer.create(<Header />);

    expect(component.toJSON()).toMatchSnapshot();
});
