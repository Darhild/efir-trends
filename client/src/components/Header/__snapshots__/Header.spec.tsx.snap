// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe Header default state 1`] = `
<div
  className="Header"
>
  <div
    className="Header-Logo"
  >
    <svg>
      logo.svg
    </svg>
  </div>
  <div
    className="Search Header-Search"
  >
    <div
      className="Icon Search-Icon"
    >
      <svg
        onClick={[Function]}
      >
        search.svg
      </svg>
    </div>
    <form
      className="Search-Form"
      onSubmit={[Function]}
    >
      <div
        className="Search-Close"
        onClick={[Function]}
      />
      <input
        className="Search-Input"
        onChange={[Function]}
        placeholder="Поиск по передачам и телеканалам"
      />
      <button
        className="Search-Submit"
        type="submit"
      >
        Найти
      </button>
    </form>
  </div>
</div>
`;
